src/app/api/[[...route]]/routes/sharehouses.ruote.ts

```diff
@@ -1,14 +1,62 @@
 import { Hono } from 'hono';
 
-export const app = new Hono();
+import prisma from '@/lib/prisma';
+import { auth } from '@/lib/auth';
+import type { IAssignedData } from '@/utils/assigned-data';
+
+const app = new Hono();
 
 export default app;
 
-app.get('/', (c) => {
-  return c.json({ message: 'Share houses route' });
-});
+app.get('/', async (c) => {
+  const session = await auth();
+
+  if (!session) return c.json({ error: 'Unauthorized' }, 401);
+
+  const shareHouses = await prisma.landlord.findUnique({
+    where: { id: session.user.id },
+    include: {
+      shareHouses: {
+        include: {
+          assignmentSheet: {
+            select: {
+              assignedData: true,
+            },
+          },
+        },
+      },
+    },
+  });
+
+  if (!shareHouses) return c.json({ error: 'Internal Server Error' }, 500);
+
+  const shareHousesWithProgress = shareHouses?.shareHouses.map((shareHouse) => {
```

Comment:
```suggestion
  const shareHousesWithProgress = shareHouses.shareHouses.map((shareHouse) => {
```

Since you added the if statement, it's better to remove the "?" from this statement.

Reply:
I removed the operator.